### ISSUE TITLE
Wrong mix of thousand and decimal separators - despite setting in lang - but without a locale.

### ISSUE THREAD

**Issue by cintaccs** [highcharts/highcharts#22424]
#### Expected behaviour
with 
```
Highcharts.setOptions({
    lang: {        
        decimalPoint: ',',
        thousandsSep: '.'
    }    
});
```
1.234.567,89

or

if switched

1,234,567.89

but never

1,234.567

#### Actual behaviour

<img width="1613" alt="Screenshot 2024-12-20 at 13 41 48" src="https://github.com/user-attachments/assets/3f43de4b-9107-4955-8b71-b4d067e52493" />

#### Product version

12.1.1

**Comment by TorsteinHonsi** [comment]
Thanks for reporting!

Unfortunately I cannot reproduce this. See https://jsfiddle.net/highcharts/n8Lp3zmw/ . The data label format doesn't enable thousands separator, but if I make it (see commented code), it still applies the correct thousands separator. I tested it in Chrome, Firefox and Safari.

Tested also with different `lang.locale` settings. The jsFiddle page sets "en" as the page language, which Highcharts picks up. 

The only variable I can think of is if your browser is of another language? Also, if you find out something, please click save on the jsFidde and send over the link.

**Comment by cintaccs** [comment]
I will! Be aware that I did not set a language but tries to "overwrite" with static separators which I found in the docs also...
https://api.highcharts.com/highcharts/lang.thousandsSep

**Comment by cintaccs** [comment]
and it used to work before 12.x

**Comment by TorsteinHonsi** [comment]
> Be aware that I did not set a language but tries to "overwrite" with static separators which I found in the docs also...

Yes, just like I did in my [fiddle](https://jsfiddle.net/highcharts/n8Lp3zmw/), which also runs v12.1.1, and everything seems to work correctly. How does my fiddle look to you? Is the number formatting correct?

You also posted a screenshot of jsFiddle. It would be great if you can save that and post the URL, so that we can see if we can reproduce the issue.

Bottom line: I can't reproduce the error, and I need to see it live in order to debug.

**Comment by TorsteinHonsi** [comment]
Hey, I know what the problem is! You weren't running v12.1.1. Here's the bug reproduced with v12.1.0. Then I've got good news, this issue was fixed in today's version 12.1.1.

**Comment by cintaccs** [comment]
I actually do run with the latest today version locally - and have the same problem but in a different use case than the Fiddle. I will investigate further and get back if needed.  Thanks for following up in any case!
